src/spatialexperiment/SpatialExperiment.py

```diff
@@ -0,0 +1,814 @@
+        return self.get_image_data()
+
+    def set_image_data(
+        self, _img_data: Optional[biocframe.BiocFrame], in_place: bool = False
+    ) -> "SpatialExperiment":
+        """Set new image data.
+
+        Args:
+            _img_data:
+                :py:class:`~biocframe.BiocFrame.BiocFrame` containing the image data, structured with the following columns:
+                    - **sample_id** (str): A string identifier for the sample to which an image corresponds.
+                    - **image_id** (str): A unique string identifier for each image within each sample.
+                    - **data** (SpatialImage): The image itself, represented as a SpatialImage object.
+                    - **scale_factor** (float): A numerical value that indicates the scaling factor applied to the image.
+
+                Image data are coerced to a
+                :py:class:`~biocframe.BiocFrame.BiocFrame`.
+
+            in_place:
+                Whether to modify the ``SpatialExperiment`` in place. Defaults to False.
+
+        Returns:
+            A modified ``SpatialExperiment`` object, either as a copy of the original or as a reference to the (in-place-modified) original.
+        """
+        img_data = _sanitize_frame(_img_data)
+        _validate_img_data(img_data)
+
+        output = self._define_output(in_place)
+        output._img_data = img_data
+        return output
+
+    def set_img_data(
+        self, img_data: biocframe.BiocFrame, in_place: bool = False
+    ) -> "SpatialExperiment":
+        """Alias for :py:meth:`~set_image_data`."""
+        return self.set_image_data(img_data=img_data, in_place=in_place)
+
+    @property
+    def img_data(self) -> biocframe.BiocFrame:
+        """Alias for :py:meth:`~get_image_data`."""
+        return self.get_image_data()
+
+    @img_data.setter
+    def img_data(self, img_data: biocframe.BiocFrame):
+        """Alias for :py:meth:`~set_image_data`."""
+        warn(
+            "Setting property 'img_data' is an in-place operation, use 'set_image_data' instead.",
+            UserWarning,
+        )
+        self.set_image_data(img_data=img_data, in_place=True)
+
+    @property
+    def image_data(self) -> biocframe.BiocFrame:
+        """Alias for :py:meth:`~get_image_data`."""
+        return self.get_image_data()
+
+    @image_data.setter
+    def image_data(self, img_data: biocframe.BiocFrame):
+        """Alias for :py:meth:`~set_image_data`."""
+        warn(
+            "Setting property 'img_data' is an in-place operation, use 'set_image_data' instead.",
+            UserWarning,
+        )
+        self.set_image_data(img_data=img_data, in_place=True)
+
+    ##############################
+    #####>> scale_factors <<######
+    ##############################
+
+    # TODO: implement getters and setters
+
+    ################################
+    ###>> OVERRIDE column_data <<###
+    ################################
+
+    def set_column_data(
```

Comment:
overriding class methods should maintain the same definition as the parent class: 

https://github.com/BiocPy/SummarizedExperiment/blob/master/src/summarizedexperiment/BaseSE.py#L540